src/browserlib/extract-algorithms.mjs

```diff
@@ -0,0 +1,649 @@
+  'end',
+  'enqueue',
+  'ensure',
+  'error',
+  'establish',
+  'execute',
+  'extend',
+  'extract',
+  'fail',
+  'fetch',
+  'finalize',
+  'find',
+  'finish',
+  'fire',
+  'gather',
+  'generate',
+  'give',
+  'handle',
+  'hand-off',
+  'increase',
+  'increment',
+  'initialize',
+  'insert',
+  'interpret',
+  'invoke',
+  'issue',
+  'jump',
+  'let',
+  'load',
+  'make',
+  'mark',
+  'match',
+  'move',
+  'multiply',
+  'navigate',
+  'paint',
+  'parse',
+  'perform',
+  'place',
+  'pop',
+  'populate',
+  'prepare',
+  'prepend',
+  'process',
+  'prompt',
+  'push',
+  'query',
+  'queue',
+  'recalculate',
+  'rectify',
+  'reference',
+  'register',
+  'reinitialize',
+  'reject',
+  'release',
+  'remove',
+  'replace',
+  'reset',
+  'resolve',
+  'resolve',
+  'restore',
+  'render',
+  'remap',
+  'report',
+  'return',
+  'run',
+  'score',
+  'scroll',
+  'send',
+  'serialize',
+  'set',
+  'shuffle',
+  'skip',
+  'sort',
+  'split',
+  'spin',
+  'start',
+  'stop',
+  'store',
+  'strip',
+  'suspend',
+  'switch',
+  'take',
+  'terminate',
+  'throw',
+  'trap',
+  'try',
+  'undisplay',
+  'unset',
+  'up-mix',
+  'update',
+  'update',
+  'upgrade',
+  'use',
+  'validate',
+  'verify',
+  'visit',
+  'wait',
+
+  'for',
+  'if',
+  'while'
+];
+
+
+/**
+ * When the step does not start with a verb, or when that verb is not followed
+ * by a white space, the following constructs help detect the actual operation.
+ */
+const stepInlineOperations = [
+  'abort all these steps',
+  'abort these steps',
+  'fire a simple event',
+  'fire an event',
+  'in parallel',
+  'reject',
+  'resolve',
+  'run the following steps',
+  'run these steps',
+  'terminate these steps',
+  /queue a( \w+)? task/i
+];
+
+
+/**
+ * Additional anchors that suggest algorithm steps
+ */
+const stepAnchors = [
+  /^⌛/,
+  'in parallel',
+  /^otherwise(\,| )/i,
+];
+
+
+/**
+ * Return the normalized text content for the given DOM element, removing all
+ * annotations
+ */
+function getTextContent(el) {
+  const clone = cloneAndClean(el);
+  return normalize(clone.textContent);
+}
+
+
+/**
+ * Return the normalized HTML content for the given DOM element, removing all
+ * annotations
+ */
+function getHTMLContent(el) {
+  // Prepare mapping table to turn relative links to absolute ones
+  // (we cannot do that once the element has been cloned because cloning
+  // removes the element from the DOM tree)
+  const relativeUrlSelector = '[href]:not([href^="http"]),[src]:not([src^="http"])';
+  const relativeToAbsolute = {};
+  const page = el.closest('[data-reffy-page]')?.getAttribute('data-reffy-page');
+  for (const linkEl of el.querySelectorAll(relativeUrlSelector)) {
+    const attr = linkEl.getAttribute('href') ? 'href' : 'src';
+    const url = new URL(page ?? window.location.href);
+    url.hash = linkEl.getAttribute(attr);
+    relativeToAbsolute[linkEl.getAttribute(attr)] = url.toString();
+  }
+
+  const clone = cloneAndClean(el);
```

Comment:
to help (somewhat) reduce the weight of the extracts, I wonder if we shouldn't apply the same [cleaning as definition content](https://github.com/w3c/reffy/blob/113e87493ccfe96e39498cb31b686d5fcf5f4a4e/src/browserlib/extract-dfns.mjs#L148-L181) (and thus migrate that code into the cloneAndClean module)

Reply:
That would reduce the weight from ~48MB to ~32MB.

I had considered it but since I did not really know how to scope the analyses we would end up doing against algorithms, I thought I'd preserve the whole thing. With definitions, we want to extract prose that can be inserted somewhere else. We need to be conservative. For analysis, the more the merrier.

There are algorithm steps that contain formulas with syntax highlighting that end up being represented through a `<c->` element, others such as Web IDL that use `<emu-val>` and `<emu-const>`, and some specs seem to integrate recent changes through `<ins>` such as [nav-speculation](https://wicg.github.io/nav-speculation/speculation-rules.html#content-security-policy-patches-match-element-to-source-list). Dropping the elements would cut words in the middle of a sentence. The ["fingerprint" image](https://fetch.spec.whatwg.org/#resolve-an-origin) used sometimes could be worth preserving as well for study.

Now, the vast majority of gains seems to come from cutting attributes. Perhaps we can just do that, but preserve elements for now.

(Also, note there's a bug in the extract-dfns function: `el.closest('[data-reffy-page]')` is unlikely going to return an element, because `el` is a detached clone. I'm surprised that does not seem to have triggered any issue so far. I'll need to fix that, as done in the extract-algorithms module)